CMakeLists.txt

```diff
@@ -26,14 +34,14 @@ if(BUILD_TESTING)
     block()
         set(INSTALL_GTEST OFF) # Disable GoogleTest installation
         set(BUILD_TESTING OFF) # Disable GoogleTest tests
+        set(BUILD_GMOCK OFF)
+        set(CMAKE_CXX_FLAGS) # FIXME: Do not propagate our project settings! CK
```

Comment:
What if your build dependencies doesn't compile with your compiler flags?

Reply:
Then the code is likely fundamentally incompatible? Except for warning flags, pretty much all other flags are detectable changes that will compile differently, leading to ODR violations, sometimes just harmless. This is true even for the standard library to some extent, although standard library implementors go to great lengths to avoid such breaks. 
Feature test macros and polyfills are common examples. Standard library implementation choice (libc++ vs libstdc++) is another. 
It's possible that some flags are peculiar to a package, or there are exceptional workarounds necessary, but by default all C++ in a link context should be compiled with the same flags.

Reply:
see https://github.com/beman-project/exemplar/pull/66#discussion_r1835769955